com.unity.ml-agents/Runtime/Timer.cs

```diff
@@ -376,14 +384,26 @@ public string DebugGetTimerString()
         /// <param name="filename"></param>
         public void SaveJsonTimers(string filename = null)
         {
-            if (filename == null)
+            try
+            {
+                if (filename == null)
+                {
+                    var activeScene = SceneManager.GetActiveScene();
+                    var timerDir = Path.Combine(Application.dataPath, "ML-Agents", "Timers");
+                    Directory.CreateDirectory(timerDir);
+
+                    filename = Path.Combine(timerDir, $"{activeScene.name}_timers.json");
+                }
+
+                var fs = new FileStream(filename, FileMode.Create, FileAccess.Write);
+                SaveJsonTimers(fs);
+                fs.Close();
+            }
+            catch (IOException)
             {
-                var fullpath = Path.GetFullPath(".");
-                filename = $"{fullpath}/csharp_timers.json";
+                // It's possible we don't have write access to the directory.
```

Comment:
Application.dataPath is read-only on mobile, but https://docs.unity3d.com/ScriptReference/Application-persistentDataPath.html seems like overkill.

Reply:
Do we need these timers on mobile ? I feel like all of these timers should be on computers only.

Reply:
No, don't think think we need them on mobile. I'm just trying to handle the the failure gracefully here. Are you suggesting we shouldn't even try to write them on mobile?

Reply:
I kind of agree with Vince, IMO we shouldn't try to save them on mobile. If for some reason the API changes to read-write later, we'll start to write things to the disk on mobile as well.

Reply:
Sounds good, I'll #ifdef them out